cmd/check_spec.go

```diff
@@ -122,58 +92,419 @@ func newCheckSpec(ctx context.Context) (*checkSpec, error) {
+		}
+
+		// block_identifier is required
+		err = validateBlockIdentifier(block)
+		if err != nil {
+			fmt.Printf("%v\n", err)
+			setValidationStatus(output, blockID, checkSpecFailure)
+		}
+
+		// coins is required
+		err = validateCoins(cs)
+		if err != nil {
+			fmt.Printf("%v\n", err)
+			setValidationStatus(output, coins, checkSpecFailure)
+		}
+	}
+
+	fmt.Println("/account/coins validated")
+	return output
+}
+
+func (cs *checkSpec) block(ctx context.Context) checkSpecOutput {
+	output := checkSpecOutput{
+		api: block,
+		validation: map[checkSpecRequirement]checkSpecStatus{
+			idempotent: checkSpecSuccess,
+			defaultTip: checkSpecSuccess,
+		},
+	}
+
+	res, fetchErr := cs.onlineFetcher.NetworkStatusRetry(ctx, Config.Network, nil)
+	if fetchErr != nil {
+		fmt.Printf("%v: unable to get network status\n", fetchErr.Err)
+		markAllValidationsFailed(output)
+		return output
 	}
 
-	if err = cs.NetworkStatus(ctx); err != nil {
-		return fmt.Errorf("%w: network status verification failed", err)
+	// multiple calls with the same hash should return the same block
+	var block *types.Block
+	var tipDiff int64 = 3
+	tip := res.CurrentBlockIdentifier
+	callTimes := 3
+
+	for i := 0; i < callTimes; i++ {
+		blockID := types.PartialBlockIdentifier{
+			Hash: &tip.Hash,
+		}
+		b, fetchErr := cs.onlineFetcher.BlockRetry(ctx, Config.Network, &blockID)
+		if fetchErr != nil {
+			fmt.Printf("%v: unable to fetch block %v\n", fetchErr.Err, blockID)
+			markAllValidationsFailed(output)
+			return output
+		}
+
+		if block == nil {
+			block = b
+		} else if types.Hash(*block) != types.Hash(*b) {
+			fmt.Printf("%v\n", errBlockNotIdempotent)
+			setValidationStatus(output, idempotent, checkSpecFailure)
+		}
 	}
 
-	if err = cs.NetworkList(ctx, cs.onlineFetcher); err != nil {
-		return fmt.Errorf("%w: online network list verification failed", err)
+	// tip shoud be returned if block_identifier is not specified
+	emptyBlockID := &types.PartialBlockIdentifier{}
+	block, fetchErr = cs.onlineFetcher.BlockRetry(ctx, Config.Network, emptyBlockID)
+	if fetchErr != nil {
+		fmt.Printf("%v: unable to fetch tip block\n", fetchErr.Err)
+		setValidationStatus(output, defaultTip, checkSpecFailure)
+		return output
 	}
 
-	if err = cs.NetworkList(ctx, cs.offlineFetcher); err != nil {
-		return fmt.Errorf("%w: offline network list verification failed", err)
+	// fetch the tip block again
+	res, fetchErr = cs.onlineFetcher.NetworkStatusRetry(ctx, Config.Network, nil)
+	if fetchErr != nil {
+		fmt.Printf("%v: unable to get network status\n", fetchErr.Err)
+		setValidationStatus(output, defaultTip, checkSpecFailure)
+		return output
+	}
+	tip = res.CurrentBlockIdentifier
+	if tip.Index-block.BlockIdentifier.Index > tipDiff {
+		fmt.Printf("%v\n", errBlockTip)
+		setValidationStatus(output, defaultTip, checkSpecFailure)
 	}
 
-	// TODO: more checks
+	fmt.Println("/block validated")
+	return output
+}
+
+func (cs *checkSpec) errorObject(ctx context.Context) checkSpecOutput {
```

Comment:
is this `errorObject` check only for `/network/status` endpoint? Other endpoints can also throw error right?

Reply:
Done.